core/drivers/crypto/caam/utils/utils_mem.c

```diff
@@ -0,0 +1,416 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   Memory management utilities.
+ *         Primitive to allocate, free memory.
+ */
+#include <arm.h>
+#include <caam_common.h>
+#include <caam_utils_mem.h>
+#include <mm/core_memprot.h>
+#include <mm/tee_mmu.h>
+#include <string.h>
+
+/*
+ * CAAM Descriptor address alignment
+ */
+#ifdef ARM64
+#define DESC_START_ALIGN	(64 / 8)
+#else
+#define DESC_START_ALIGN	(32 / 8)
+#endif
+
+/*
+ * Check if pointer p is aligned with align
+ */
+#define IS_PTR_ALIGN(p, align) (((uintptr_t)(p) & ((align) - 1)) == 0)
+
+/*
+ * Check if size is aligned with align
+ */
+#define IS_SIZE_ALIGN(size, align)                                             \
+	({                                                                     \
+		__typeof__(size) _size = (size);                               \
+		__typeof__(size) _sizeup = 0;                                  \
+									       \
+		_sizeup = ROUNDUP(_size, align);                               \
+		(_sizeup == _size) ? 1 : 0;                                    \
+	})
+
+/*
+ * Read the system cache line size.
+ * Get the value from the ARM system configration register
+ */
+static uint32_t read_cacheline_size(void)
+{
+	uint32_t value = 0;
+
+#ifdef ARM64
+	value = read_ctr_el0();
+#else
+	value = read_ctr();
+#endif
+	value = CTR_WORD_SIZE <<
+		((value >> CTR_DMINLINE_SHIFT) & CTR_DMINLINE_MASK);
+	MEM_TRACE("System Cache Line size = %u bytes", value);
```

Comment:
s/`%u`/`%"PRIx32"`/

Reply:
no in this case we would like the decimal number of bytes